src/Datadog.Trace/Tracer.cs

```diff
@@ -309,20 +316,20 @@ public Span StartSpan(string operationName, ISpanContext parent = null, string s
             if (parent is SpanContext parentSpanContext)
             {
                 traceContext = parentSpanContext.TraceContext ??
-                               new TraceContext(this)
+                               new TraceContext(_agentWriter.WriteTrace, Sampler.GetSamplingPriority)
```

Comment:
This code is now allocating two delegates for every trace. Consider caching the delegates and reusing them.